File: BAEKJOON/스택/10828_스택/10828.py
# readline으로 안읽으면 시간초과남

import sys

sys.stdin = open('input.txt', 'r')

N = int(input())

stack = []

for _ in range(N):
    cmds = sys.stdin.readline().split()
    cmd = cmds[0]

    if cmd == 'push':
        stack.append(int(cmds[1]))
    elif cmd == 'pop':
        if len(stack) > 0:
            print(stack.pop())
        else:
            print(-1)
    elif cmd == 'size':
        print(len(stack))
    elif cmd == 'empty':
        print(1) if len(stack) == 0 else print(0)
    elif cmd == 'top':
        if len(stack) > 0:
            print(stack[-1])
        else:
            print(-1)

File: BAEKJOON/10828_스택/10828.js
const fs = require('fs');
const fileStream = fs.createReadStream('input.txt');
const readline = require('readline');

const rl = readline.createInterface({
  input: fileStream,
  output: process.stdout,
});

let input = [];

class Stack {
  constructor(N) {
    this.stack = Array(N).fill(0);
    this.top = -1;
  }

  push(num) {
    this.top += 1;
    this.stack[this.top] = num;
  }

  pop() {
    if (this.top > -1) {
      const value = this.stack[this.top];
      this.stack[this.top] = 0;
      this.top -= 1;
      return value;
    } else {
      return -1;
    }
  }

  size() {
    return this.top + 1;
  }

  empty() {
    return this.top > -1 ? 0 : 1;
  }

  getTop() {
    if (this.top > -1) {
      return this.stack[this.top];
    }
    return -1;
  }
}

function stackProcess(N, cmds) {
  const stack = new Stack(N);
  let result = '';
  cmds.forEach((cmd) => {
    cmd = cmd.split(' ');
    if (cmd[0] === 'push') {
      stack.push(Number(cmd[1]));
    } else if (cmd[0] === 'top') {
      result += stack.getTop() + '\n';
    } else if (cmd[0] === 'pop') {
      result += stack.pop() + '\n';
    } else if (cmd[0] === 'size') {
      result += stack.size() + '\n';
    } else if (cmd[0] === 'empty') {
      result += stack.empty() + '\n';
    }
  });

  return result;
}

rl.on('line', (line) => {
  input.push(line);
}).on('close', () => {
  console.log(stackProcess(Number(input[0]), input.slice(1)));
  process.exit();
});
